maskrcnn_benchmark/engine/trainer.py

```diff
@@ -107,6 +108,37 @@ def do_train(
             )
         if iteration % checkpoint_period == 0:
             checkpointer.save("model_{:07d}".format(iteration), **arguments)
+            if data_loader_val is not None:
+                meters_val = MetricLogger(delimiter="  ")
+                synchronize()
+                with torch.no_grad():
+                    for idx_val, (images_val, targets_val, _) in enumerate(data_loader_val):
+                        images_val = images_val.to(device)
+                        targets_val = [target.to(device) for target in targets_val]
+                        loss_dict = model(images_val, targets_val)
+                        losses = sum(loss for loss in loss_dict.values())
+                        loss_dict_reduced = reduce_loss_dict(loss_dict)
+                        losses_reduced = sum(loss for loss in loss_dict_reduced.values())
+                        meters_val.update(loss=losses_reduced, **loss_dict_reduced)
```

Comment:
This line records batch'loss of val-set using current train iteration model, right?
So, if the purpose is to check our model is over fitting or not, we need to calculate the average loss of val-set using current train iteration model. And use this average loss to decide early stop.

Reply:
Yes, global average loss is needed and `MetricLogger` class calculates it:
https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/utils/metric_logger.py#L35-L37